### PR TITLE
Add recursion depth for `FeatureTreeNode::getChildren`

### PR DESCRIPTION
**Pull request by s9latimm**
Add a depth parameter and update documentation.

### CODE REVIEW

File: include/vara/Feature/FeatureTreeNode.h [hunk 1]
```diff
@@ -76,10 +76,19 @@ class FeatureTreeNode : private llvm::DGNode<FeatureTreeNode, FeatureTreeNode> {
...
+  ///  structure.
+  ///
+  /// \tparam[in] T Type of expected nodes.
+  ///
+  /// \param[in] Depth Maximal recursion depth to which nodes, which does not
+  ///  match T are expanded to their children. A depth of one means, only
+  ///  children of this node are considered and disables expansions.
+  ///
+  /// \return Set of children nodes.
+  template <typename T> llvm::SmallSet<T *, 3> getChildren(int Depth = -1) {
```
Comment (vulder) [inline include/vara/Feature/FeatureTreeNode.h]: Please add a default template parameter to FeatureTreeNode

**Review by vulder — CHANGES_REQUESTED** [review]
Could you please add 1-2 tests for edge cases, like when the depth given is deeper than the tree itself.

**Review by vulder — APPROVED** [review]
